ibis/tests/all/test_vectorized_udf.py

```diff
@@ -1,26 +1,73 @@
+import pandas as pd
 import pytest
 
+import ibis
 import ibis.common.exceptions as com
 import ibis.expr.datatypes as dt
 from ibis.tests.backends import Pandas, PySpark
-from ibis.udf.vectorized import elementwise
-
+from ibis.udf.vectorized import elementwise, reduction
 
 pytestmark = pytest.mark.udf
 
 
+@elementwise(input_type=[dt.double], output_type=dt.double)
+def add_one(s):
+    return s + 1
+
+
+@reduction(input_type=[dt.double], output_type=dt.double)
+def calc_mean(s):
+    return s.mean()
+
+
 @pytest.mark.only_on_backends([Pandas, PySpark])
 @pytest.mark.xfail_unsupported
 def test_elementwise_udf(backend, alltypes, df):
-    @elementwise(input_type=[dt.double], output_type=dt.double)
-    def add_one(s):
-        return s + 1
-
     result = add_one(alltypes['double_col']).execute()
     expected = add_one.func(df['double_col'])
     backend.assert_series_equal(result, expected, check_names=False)
 
 
+@pytest.mark.only_on_backends([Pandas, PySpark])
+@pytest.mark.xfail_unsupported
+def test_reduction_udf(backend, alltypes, df):
+    result = calc_mean(alltypes['double_col']).execute()
+    expected = df['double_col'].agg(calc_mean.func)
+    assert result == expected
+
+
+@pytest.mark.only_on_backends([Pandas, PySpark])
+@pytest.mark.xfail_unsupported
+def test_reduction_udf_aggregate(backend, alltypes, df):
```

Comment:
Do you have a test for groupby aggregate with UDF?

Reply:
Oops, was missing this.

I've now actually moved the UDF aggregation and window tests to [`test_aggregation.py`](https://github.com/ibis-project/ibis/blob/186aa0b10d10e99706aa75de512a437846e7c09b/ibis/tests/all/test_aggregation.py) and [`test_window.py`](https://github.com/ibis-project/ibis/blob/186aa0b10d10e99706aa75de512a437846e7c09b/ibis/tests/all/test_window.py) so that we can just reuse the tests on both non-UDF and UDF tests parameters.

[`test_aggregation.py`](https://github.com/ibis-project/ibis/blob/186aa0b10d10e99706aa75de512a437846e7c09b/ibis/tests/all/test_aggregation.py) tests grouped and non-grouped aggregate.